web3/middleware/fixture.py

```diff
@@ -6,9 +6,11 @@ def construct_fixture_middleware(fixtures):
     def fixture_middleware(make_request, web3):
         def middleware(method, params):
             if method in fixtures:
-                return {
-                    'result': fixtures[method],
-                }
+                response = fixtures[method]
+                if callable(response):
+                    return response(method, params)
+                else:
+                    return response
```

Comment:
Why change the fixture API here? The old one that automatically inserts the value with a `{'result': val}` seemed fine.

Leaving the old API even seems to make  `tests/core/gas-strategies/test_time_based_gas_price_strategy.py` simpler by removing the need for the `wrap_result` decorator.

Reply:
Originally this was a "theoretically I bet it would be useful to be able to return error responses".

Now with the bug in the caching branch, I'm planning to actually use that functionality so the ability to return any type of json is desired.  I'm open to suggestions for cleaning up the API for the common use case.